src/mmw/js/src/modeling/views.js

```diff
@@ -66,6 +67,17 @@ var ModelingHeaderView = Marionette.LayoutView.extend({
 
     onShow: function() {
         this.reRender();
+    },
+
+    saveAfterLogin: function(user, guest) {
+        if (!guest && this.model.isNew()) {
+            var user_id = user.get('id');
+            this.model.set('user_id', user_id);
```

Comment:
After doing some more investigation, I find that [modeling/models.js#L66-68](https://github.com/WikiWatershed/model-my-watershed/blob/develop/src/mmw/js/src/modeling/models.js#L66-68) checks the `user_id` before saving to ensure that users can save only their own projects, and not another user's public project. For a logged in user creating a new project, the `user_id` is set during initialization in [modeling/models.js#L53](https://github.com/WikiWatershed/model-my-watershed/blob/develop/src/mmw/js/src/modeling/models.js#L53). However, for projects created before a user is logged in, that field defaults to `0`. After logging in, the `App.user.id` is different from `model.get('user_id')` and the save would not happen, thus necessitating this bit of code.

I believe this check for `user_id` before saving reduces erroneous AJAX requests. I don't understand the code on the server side completely due to my unfamiliarity with Django Rest Framework, but I hope that [modeling/views.py#L47](https://github.com/WikiWatershed/model-my-watershed/blob/develop/src/mmw/apps/modeling/views.py#L47) checks against the user on the server side as well.